test/built-ins/Iterator/from/iterable-primitives.js

```diff
@@ -37,3 +37,17 @@ assert.throws(TypeError, function () {
 assert.compareArray(Array.from(Iterator.from(new Number(5))), [0, 1, 2, 3, 4]);
 
 assert.compareArray(Array.from(Iterator.from('string')), ['s', 't', 'r', 'i', 'n', 'g']);
+
+const originalStringIterator = String.prototype[Symbol.iterator];
+let observedType;
+Object.defineProperty(String.prototype, Symbol.iterator, {
+  get(){
+    'use strict';
+    observedType = typeof this;
```

Comment:
looking over https://github.com/tc39/proposal-iterator-helpers/pull/281/files#, i think this still won't be the case, because GetMethod calls GetV which calls ToObject.